core/src/main/java/org/axonframework/messaging/Message.java

```diff
@@ -91,4 +93,28 @@
      * @return a copy of this message with the given MetaData
      */
     Message<T> andMetaData(Map<String, ?> metaData);
+
+    /**
+     * Serialize the payload of this message using given {@code serializer}, using given
```

Comment:
Suggestion: I'd rewrite the first sentence to:
`Serialize the payload of this message to the {@code expectedRepresentation} using the given {@link Serializer}.`